httpParser.cpp

```diff
@@ -0,0 +1,122 @@
+#include "httpParser.hpp"
+#include <sstream>
+
+httpParser::httpParser(std::map<int, Request>::iterator& pair, const WebservConfigStruct sett): req(pair), settings(sett) {
+	beheader(req->second);
+	if (req->second.HeaderBuffer.empty() == true) {
+		return;
+	}
+	if (req->second.BodyBuffer.empty() == true) {
+		return;
+	}
+	handleBody(req->second);
+}
+
+void httpParser::beheader(Request& request) {
+	size_t neck = request.RequestBuffer.find("\r\n\r\n");
+	if (neck == request.RequestBuffer.npos) {
+		std::cerr << "invalid/incomplete header: \n" << std::endl;
+		return;
+	} else if (request.HeaderBuffer.empty() == true){
+		request.HeaderBuffer = request.RequestBuffer.substr(0, neck + 4);
+		request.RequestBuffer.erase(0,neck+4);
+		GetRequestType(request);
+		GetRequestedPath(request);
+		decapitalizeHeaderFields(request.HeaderBuffer);
+		extractHeaderFields(request);
+	}
+	request.BodyBuffer.append(request.RequestBuffer.substr(0, request.RequestBuffer.size()));
+	request.RequestBuffer.erase(0,request.RequestBuffer.size());
+	if (request.Body.size() > settings.client_max_body_size) {
+		this->req->second.Integrity = BODY_TOO_BIG;
+	}
+	//std::cout << request.HeaderBuffer << std::endl;
+}
+
+void httpParser::GetRequestType(Request& request) {
```

Comment:
What if there is a request with an invalid method in the first line of the header, but e.g. GET is used later in the HeaderBuffer? This should also count as invalid but the current implementation will accept it

Reply:
Changed code to first extract the first line of the headerBuffer and search for methods in that.